### ISSUE TITLE
A quick way to builtin serializer for my own library?

### ISSUE THREAD

**Issue by RinOrz** [Kotlin/kotlinx.serialization#1693]
Add serializers support for my own library, but do not add runtime dependency for kotlinx.serialization. What should I do?

**Comment by RinOrz** [comment]
When I mark @Serializable in library and then try to serialize it in another library, I get:

> Consider to add one of the following dependencies: 
>  - ktor-client-gson
>  - ktor-client-json
>  - ktor-client-serialization

**Comment by sandwwraith** [comment]
> add serializers support for my own library, but do not add runtime dependency of kotlinx.serialization

That's not possible. You either need dependency on `kotlinx-serialization-core` to write serializer manually, or both dependency and `plugin("kotlinx-serialization")` to generate serializers with `@Serializable` annotation

**Comment by RinOrz** [comment]
@sandwwraith I saw the `kotlinx.serialization` used with `compileOnly` in https://github.com/Kotlin/kotlinx-datetime/blob/44b8ee3caad84919f539fd1a44e124965e975877/core/build.gradle.kts#L156. Is this the right approach?

https://github.com/Kotlin/kotlinx-datetime/blob/44b8ee3caad84919f539fd1a44e124965e975877/core/jvm/src/TimeZoneJvm.kt#L17

**Comment by sandwwraith** [comment]
@RinOrz yeah, it may work. Although you should have in mind that accessing any of serializaiton-related functions in your library will result in `NoClassDefFoundError` if serialization runtime would be missing from classpath

**Comment by RinOrz** [comment]
yeah, I understand this, but the class marked `@Serializable` probably not throw an exception? this is also my concern

**Comment by RinOrz** [comment]
I am not very clear about the behind the scenes work of the compiler plugin. I'm just a little worried that the annotated class will need the serialization runtime during class initialization

**Comment by sandwwraith** [comment]
I think there were some `<clinit>` blocks with runtime references, but you may try
